### PR TITLE
8257910: [JVMCI] Set exception_seen accordingly in the runtime.

### PR DESCRIPTION
**Pull request by mur47x111**
Update exception_seen flag in the MDO when no exception handler is found for caller during deoptimization, or when exception is thrown in the runtime.

Plus, `-Xlog:exceptions=info` code and `AbortVMOnException` are adjusted in C1 runtime and jvmci runtime to be consistent with C2, i.e., before `JvmtiExport::can_post_on_exceptions` deoptimization and fast continuation when compiled handler is found.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed

### Issue
 * [JDK-8257910](https://bugs.openjdk.java.net/browse/JDK-8257910): [JVMCI] Set exception_seen accordingly in the runtime.

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1713/head:pull/1713`
`$ git checkout pull/1713`

### CODE REVIEW

File: src/hotspot/share/runtime/sharedRuntime.cpp [hunk 1]
```diff
@@ -599,6 +599,28 @@ void SharedRuntime::throw_and_post_jvmti_exception(JavaThread *thread, Handle h_
     address bcp = method()->bcp_from(vfst.bci());
     JvmtiExport::post_exception_throw(thread, method(), bcp, h_exception());
   }
+
+#if INCLUDE_JVMCI
+  if (EnableJVMCI) {
```
Comment (vnkozlov) [inline src/hotspot/share/runtime/sharedRuntime.cpp:604]: I think this should also check `&& UseJVMCICompiler`

**Review by vnkozlov — CHANGES_REQUESTED** [review]

File: src/hotspot/share/jvmci/jvmciRuntime.cpp [hunk 2]
```diff
@@ -267,6 +267,20 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* t
...
+    ResourceMark rm;
+    stringStream tempst;
+    assert(cm->method() != NULL, "Unexpected null method()");
+    tempst.print("compiled method <%s>\n"
+                 " at PC" INTPTR_FORMAT " for thread " INTPTR_FORMAT,
+                 cm->method()->print_value_string(), p2i(pc), p2i(thread));
+    Exceptions::log_exception(exception, tempst.as_string());
+  }
+  // for AbortVMOnException flag
+  NOT_PRODUCT(Exceptions::debug_check_abort(exception));
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciRuntime.cpp:282]: All other places calls debug_check_abort() in product too. It seems 8136577 changes missed JVMCI:
https://github.com/openjdk/jdk/commit/a25ce80e782a73390ae1109250dfa6fe6a48fcbe
Please, update to call it in product for JVMCI too. Note, AbortVMOnException is diagnostic flag now.

File: src/hotspot/share/c1/c1_Runtime1.cpp [hunk 3]
```diff
@@ -538,6 +538,20 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* t
   }
 #endif
 
+  // debugging support
+  // tracing
+  if (log_is_enabled(Info, exceptions)) {
+    ResourceMark rm;
+    stringStream tempst;
+    assert(nm->method() != NULL, "Unexpected NULL method()");
+    tempst.print("compiled method <%s>\n"
```
Comment (vnkozlov) [inline src/hotspot/share/c1/c1_Runtime1.cpp:547]: Can you change message:
"C1 compiled method <%s>\n"

File: src/hotspot/share/jvmci/jvmciRuntime.cpp [hunk 4]
```diff
@@ -267,6 +267,20 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* t
   }
 #endif
 
+  // debugging support
+  // tracing
+  if (log_is_enabled(Info, exceptions)) {
+    ResourceMark rm;
+    stringStream tempst;
+    assert(cm->method() != NULL, "Unexpected null method()");
+    tempst.print("compiled method <%s>\n"
```
Comment (vnkozlov) [inline src/hotspot/share/jvmci/jvmciRuntime.cpp:276]: Can you change message to?:
"JVMCI compiled method <%s>\n"

**Review by vnkozlov — CHANGES_REQUESTED** [review]

**Review by vnkozlov — CHANGES_REQUESTED** [review]
Looks good. Unfortunately it missed JDK16 fork but because it is P3 bug it can be fixed there without approval for now (RDP1 phase).
Please, create PR based on https://github.com/openjdk/jdk16 (create your fork for it) and I will approve it. After push the fix will be auto-forwarded into JDK 17 (current  https://github.com/openjdk/jdk)

**Comment by mur47x111** [comment]
Thanks @vnkozlov ! Created https://github.com/openjdk/jdk16/pull/2 and took the opportunity to squash commits.

**Pull request closed without merging**